simulator_libary/generator/FiberBoxPacketGenerator.cs

```diff
@@ -0,0 +1,41 @@
+﻿using simulator_libary.generator;
+using System.Collections.Generic;
+
+namespace simulator_libary
+{
+    public class FiberBoxPacketGenerator<IcdType> : BasePacketGenerator<IcdType> where IcdType : IParameterIcd
+    {
+        public FiberBoxPacketGenerator(string json):base(json){}
+        public override void GenerateByteArray(List<IcdType> icdRows, ref byte[] finalSequence, List<IcdType> errorLocations)
```

Comment:
you have same lines in FiberboxPacketGenerator (lines 19-25,30-36) and in FlightBoxPacketGenerator (lines 14-28), put them in abstract class